trieste/models/model_interfaces.py

```diff
@@ -175,22 +94,22 @@ def predict(self, query_points: QueryPoints) -> Tuple[ObserverEvaluations, Tenso
     def sample(self, query_points: QueryPoints, num_samples: int) -> ObserverEvaluations:
         return self.model.predict_f_samples(query_points, num_samples)
 
+    def optimize(self, dataset: Dataset):
+        self.optimizer.optimize(self.model, dataset)
```

Comment:
since the optimizer doesn't access private on the model, we should be able to completely decouple the optimizers and models. What do you think about doing that?

Reply:
this may be a separate option from merging the two methods

Reply:
@joelberkeley ^^^

Reply:
you could do this like
```
class _WithOptimizer(GPflowPredictor):
    def __init__(self, optimizer = None):
        self._optimizer = optimizer

    def optimize(self):
        self._optimizer.optimize(self.model)
```
then inherit from this class instead

Reply:
with regards to testing, I think it's OK to change lots of tests if the design asks for it

Reply:
I think the question is more about delaying the delivery of this PR or simply move it to another ticket.

Reply:
sorry i've conflated conversations here. my suggestion was for something else. I'm ok to leave the optimizers as part of the models. I'll put the other suggestion where I meant to put it